test/libevmasm/Optimiser.cpp

```diff
@@ -967,6 +967,31 @@ BOOST_AUTO_TEST_CASE(peephole_swap_comparison)
 	}
 }
 
+BOOST_AUTO_TEST_CASE(peephole_truthy_and)
+{
+  AssemblyItems items{
+    AssemblyItem(Tag, 1),
+    u256(1),
```

Comment:
Ah sorry, I just saw that now. This might be a bad test case, since all items here are constants and thus a different (perhaps future) peephole optimizer routine might do the same job. Can you replace this line by e.g. `Instruction::BALANCE`?